modules/accounts/email.nix

```diff
@@ -508,6 +508,13 @@ in {
       default = { };
       description = "List of email accounts.";
     };
+
+    order = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description =
+        "Order of email accounts from [](#opt-accounts.email.accounts).";
+    };
```

Comment:
I think a nicer approach would be to add an `order` field to each account, and sort the accounts based on that field. Because with the current implementation I see a couple of issues:

* If you add an account to `accounts.email.accounts` but forget to add it to `accounts.email.order`, that account will be ignored by aerc.
* It is possible to make a typo in the account name, or have an entry in `accounts.email.order` that does not exist in `accounts.email.accounts`. Both of which will cause an error.

Of course there's also a downside: multiple accounts can have the same `order` value. But that's much easier to debug, and we could even add an assertion to prevent that from happening.

Reply:
Thanks for the review @olmokramer! To address your issues:

> If you add an account to accounts.email.accounts but forget to add it to accounts.email.order, that account will be ignored by aerc.

I think you misunderstand, any accounts not included in `order` will be included in the aerc account config by virtue of the `unorderedAccounts` variable.

If only some account names are in the `order` list then they will be included in that order after the primary account, and any additional accounts will be included with the previous behavior.

> It is possible to make a typo in the account name, or have an entry in accounts.email.order that does not exist in accounts.email.accounts. Both of which will cause an error.

This is true. Perhaps some assertions could make this easier to debug?

> a nicer approach would be to add an order field to each account

I'm not sure I agree. For me this would be less ergonomic, in that if one were to re-order an account you would need to change this index in multiple places, as opposed to moving a list entry.

Reply:
Hey @RyanGibb thanks for the reply. You're right, I missed the `unorderedAccounts` so forget about that.

And I hadn't really thought about the need to reorder accounts, I can see that being a pain with my proposed method. I don't use Home Manager's email infrastructure, so I don't mind which direction this takes in the end.

Finally, I think some assertions would go a long way to prevent the errors I had in mind in my previous comment, or at least make them much easier to debug.

Reply:
Cool! I've added an assertion that fails with:
```
       Failed assertions:
       - ryan profile: 'example@example.com' is present in config.accounts.email.orders but not in config.accounts.email.accounts.
```